### PR TITLE
Deprecate device.decompose_operation in cirq_ionq.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Yet more of #4744 .

### CODE REVIEW

File: cirq-ionq/cirq_ionq/ionq_devices.py [hunk 1]
```diff
@@ -78,42 +81,52 @@ def validate_operation(self, operation: cirq.Operation):
...
-            lambda op, _: cirq.decompose_once(op) if type(op.gate) == cirq.PhasedXPowGate else op,
-        ).all_operations()
+        return decompose_to_device(operation)
+
+
+def decompose_to_device(operation: cirq.Operation, atol: float = 1e-8) -> cirq.OP_TREE:
+    if operation in _VALID_GATES:
+        return operation
+    assert cirq.has_unitary(operation), (
+        f'Operation {operation} that is not available on the IonQ API nor does it have a '
```
Comment (dabacon) [inline cirq-ionq/cirq_ionq/ionq_devices.py:98]: remove "that"

File: cirq-ionq/cirq_ionq/ionq_devices.py [hunk 2]
```diff
@@ -16,6 +16,23 @@
 from typing import AbstractSet, Sequence, Union
 
 import cirq
+from cirq import _compat
+
+
+_VALID_GATES = cirq.Gateset(
```
Comment (dabacon) [inline cirq-ionq/cirq_ionq/ionq_devices.py:22]: adding to the import time I see :)

**Review by dabacon — APPROVED** [review]
LGTM

**Comment by tanujkhattar** [comment]
The original `decompose_operation` implementation was a sub-routine used to convert the constructed circuit to the native target gateset. The newly added `decompose_to_device` is a method that just converts individual operations. 

We should ideally create a new cirq transformer (aka optimizer) for converting circuits to the ionq native gatesets and use that as a recommendation to replace the deprecated `decompose_operation`. 

This is tracked in https://github.com/quantumlib/Cirq/issues/4901

**Comment by MichaelBroughton** [comment]
Right, although I don't think you or I should work on implementing this. Ideally someone from ionq will come along and do this.

**Pull request merged**